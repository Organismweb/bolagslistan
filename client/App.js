// @flow
import React, { Component } from 'react';
import styled from 'styled-components';
import './components/_settings/_base';
import Header from './components/organisms/Header';
import Sidebar from './components/organisms/Sidebar';
import Main from './components/organisms/Main';

type State = {
  text: string,
};

class App extends Component<void, State> {
  // State using Class properties transform
  state = {
    text: 'asasdsadd',
  };
  // How to write functions to get correct scope of this.
  exampleFunction = () => <div>{this.state.text}</div>;
  render() {
    return (
      <AppContainer>
        <Header />
        <Sidebar />
        <Main>Main content goes here</Main>
      </AppContainer>
    );
  }
}

const AppContainer = styled.div`
  display: grid;
  grid-template-areas:
    'header header header header'
    'sidebar main main main'
    'sidebar main main main';
  grid-template-columns: 300px 1fr 1fr 1fr;
  grid-template-rows: 70px 1fr 1fr;
  height: 100vh;
  width: 100vw;
  overflow: hidden;
`;

export default App;
