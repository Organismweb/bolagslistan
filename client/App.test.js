import React from 'react';
import App from './App';
import { mount, shallow } from 'enzyme';

test('app component renders without exploding', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
