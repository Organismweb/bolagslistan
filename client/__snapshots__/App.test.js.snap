// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component renders without exploding 1`] = `
<React.Fragment>
  <div>
    asasdsadd
  </div>
</React.Fragment>
`;
